lumicks/pylake/kymo.py

```diff
@@ -237,6 +246,26 @@ def plot_with_time_histogram(self, color_channel, pixels_per_bin=1, hist_ratio=0
         ax_hist.set_ylabel("counts")
         ax_hist.set_title(self.name)
 
+    def downsampled_by(self, downsampling_factor, reduce=np.sum):
```

Comment:
* Needs a docstring.
* Rename `downsampling_factor` to just `factor`. It's redundant otherwise. Consider: `kymo.downsampled_by(downsampling_factor=2)` -- it's repeating itself. This is why it's good to use named args in the docs: both as a good first example for users and to see if there are any naming redundancies. (Also: `factor` makes it consistent with `Slice.downsampled_by()`.
* Think about how this API would evolve if the space dimension also needed to be downsampled (only think about it for now).

Reply:
To your last point, that reads as if you'd _want_ to prefix the scaling factors, in case you end up with one for each dimension.

Reply:
Yes. Good point.

Regarding your third point, maybe it'd be good to just be explicit. How about `time_factor`? It does mean breaking with `Channel` (which uses `factor`), but then we future proof it in case we end up supporting spatial downsampling.

Reply:
I'd be ok with `time_factor`. Kymos are inherently higher dimensional than a `Channel` so makes sense that arguments might not be able to be named exactly the same.

Reply:
👍 👍 👍